libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/Instances.hs

```diff
@@ -1528,3 +1557,73 @@ instance IsConwayUniv fn => HasSpec fn (Pulser StandardCrypto)
 
 instance HasSimpleRep (CertsEnv (ConwayEra StandardCrypto))
 instance IsConwayUniv fn => HasSpec fn (CertsEnv (ConwayEra StandardCrypto))
+
+-- CompactForm
+
+class CoercibleLike a b where
```

Comment:
I added this constraint to the methods instead, because superclasses tend to cause problems.

Reply:
> because superclasses tend to cause problems.

What kind of problems? First time I hear this.

You solve a totally different problem by placing a constraint on a type class function instead of placing it as a superclass. Putting constraint as a super class is almost always better than on the class function. 

If you place it on a function, then only that function is unusable whenever a type does not have the constraint. Eg.

```haskell
class Foo a where
  foo :: Num a => a -> Integer
  bar :: a -> String

instance Foo Bool where
  foo _ = error "Impossible: Bool is not a Num"
  bar x = if x then "Yes" else "No"

instance Foo Int where
  foo = toInteger
  bar = show
```

So, it essentially can be used to have partial interface for a type. On the other hand, if the constraint is a super class, then the instance as a whole is impossible. We definitely want the latter, otherwise what is the point of adding `Coerce` constraint at all. This is the latter:

```haskell
class Num a => Foo a where
  foo :: a -> Integer
  bar :: a -> String

instance Foo Int where
  foo = toInteger
  bar = show
```
You see, now instance for `Bool` is impossible. The added benefit of a superclass is that you don't need to specify redundant constraints at the call site:

```haskell
baz :: Foo a => a -> Integer
baz x = foo (x + 6)
```

vs the constraint on the function:

```haskell
baz :: (Num a, Foo a) => a -> Integer
baz x = foo (x + 6)
```

This latter benefit was the actual reason why I made a suggestion. So, please either remove the `Coerce` constraint altogether or put it as a superclass.